src/sshfs_mount/sftp_server.cpp

```diff
@@ -242,10 +245,26 @@ void check_sshfs_status(mp::SSHSession& session, mp::SSHProcess& sshfs_process)
     }
 }
 
-auto create_sshfs_process(mp::SSHSession& session, const std::string& source, const std::string& target)
+auto create_sshfs_process(mp::SSHSession& session, mp::SshfsMount::SshfsPkgType sshfs_pkg_type,
+                          const std::string& source, const std::string& target)
 {
-    auto sshfs_process = session.exec(fmt::format(
-        "sudo sshfs -o slave -o nonempty -o transform_symlinks -o allow_other :\"{}\" \"{}\"", source, target));
+    std::string options{"-o slave -o transform_symlinks -o allow_other"};
+    std::string sshfs_exec{"sshfs"};
+
+    if (sshfs_pkg_type == mp::SshfsMount::SshfsPkgType::debian)
+    {
+        // FIXME: Need query the libfuse version to see if `nonempty` is valid.
+        // The option was removed in libfuse 3.0
+        options += " -o nonempty";
```

Comment:
What happens if it's removed? Could we try, fail, and try again without that option instead?

Reply:
If what is removed?  The option?  The `multipass-sshfs` snap?  The packaged version if `libfuse`?

Reply:
The option removed :)

Oh so now we need `libfuse` on the system, too?

Reply:
Yes, `sshfs` requires `libfuse`, but I redid how to detect if the option is needed or not.